cmd/server/config/config.go

```diff
@@ -122,6 +92,19 @@ func LoadConfig(path string, v *Config) error {
 	return v.Validate()
 }
 
+func SaveConfig(path string, v *Config) error {
+	if v == nil {
+		return fmt.Errorf("config cannot be nil")
+	}
+
+	yml, err := yaml.Marshal(v)
+	if err != nil {
+		return fmt.Errorf("could not marshal provided config")
+	}
+
+	return os.WriteFile(path, yml, os.ModePerm)
```

Comment:
Permission bits on Windows do not work properly for files created with Go, so `os.ModePerm` is no better than `600` - ultimately windows uses access control lists to limit who can use the file. There is work scheduled in the coming releases for fixing many of the ACLs on files created by wins and rancher, and I'd like to handle this file as a part of that work instead of in this PR